### ISSUE TITLE
Thread keeps running for a while after service is stoped

### ISSUE THREAD

**Issue by hunezhaw** [Topshelf/Topshelf#249]
Hi all
Im using Topshelf to implement a self hosted MVC application. Generally things work find. However I have one question. When I stop the service either in the MMC or command line Windows gives back that the service is stopped. However when I check in task manager I see that the exe is still running for another 5-10 seconds. This results in two problems: 
* When I hit restart, I see two threads running and sometimes I see in the log, that the first one is still shutting down, while the second one is starting. This is not really a good situation. At a given time only one service should be running.
* Deployment is not stright forwards. I first have to stop the service and then wait for a moment (check task manager) until I run the install command, because certain files are still locked by the stopping service. Unfortunatly Windows reports the service to be stoped even so the thread is still running.

Is there any way to tell Topshelf to wait telling Windows the service is stopped until the service is really terminated?

Thanks
Philipp

**Comment by TravisTheTechie** [comment]
I don't think I've see this problem with hitting restart in the services panels. What version of Windows are you running on?

For upgrades, I believe this is an issue with the [SCM](https://en.wikipedia.org/wiki/Service_Control_Manager), I don't have a great work around. When scripting upgrades I've just worked a few second sleep in after shutting a service down to make sure it's really done.

**Comment by hunezhaw** [comment]
I can observe this on Windows 2008 R2 and also Windows 2012 R2. I needed to tell my admins that restarting the service is: a) stop the service, b) look at task manager until exe is stoped, c) restart the service (not intuitive and not as it should be). If they do not do this I may have two instances of the service running. For upgrades I also added some wait time, but it takes up to 10 seconds until the exe really terminates.
